frontend/src/components/common/Link.tsx

```diff
@@ -77,5 +77,39 @@ export default function Link(props: React.PropsWithChildren<LinkProps | LinkObje
     }
   }
 
+  if ('kubeObject' in props && kubeObject) {
+    {
+      /* NOTE: there is an issue where TS would throw an error for using KubeObject<any> along with the return undefined, this seems to fix it
+       * if we were to use dispatch(setSelectedResource(kubeObject!)); the error for returning kubeObject<any> & void would appear
+       */
+    }
+
+    const kubeJSON = kubeObject.jsonData;
+    return (
+      <>
+        {drawerEnabled === true && props.kubeObject ? (
+          <MuiLink
+            onClick={() => {
+              if (drawerEnabled) {
+                dispatch(setSelectedResource(kubeJSON!));
+                window.history.pushState(
```

Comment:
please use react-router history instead, mutating window.history will not play nicely with our existing router

Reply:
I previously used a separate component that would display the "drawer mode" links so that when clicking on a link it would not direct the user to the details page itself but rather the "/drawer" url variant

I recall moving away from that solution because the goal was to be able to send links easily without having it be dependent on having drawer mode enabled

I have tried using history.push and history.replace from the useHistory but both seem to force a page navigation, any other ideas in mind for this situation or should I just remove the URL changes entirely?